selendroid-server/src/main/java/io/selendroid/server/model/AndroidNativeElement.java

```diff
@@ -254,18 +256,73 @@ public String getText() {
   public void click() {
     waitUntilIsDisplayed();
     scrollIntoScreenIfNeeded();
+
+    final View view = getView();
+
+    final int[] xy = new int[2];
+
+    if (Build.VERSION.SDK_INT < 19 || view.isLaidOut()) {
+      if (Build.VERSION.SDK_INT < 19) {
+        try {
+          // This is needed for recalculation of location when we cannot
+          // explicitly check if the view's been laid out
+          Thread.sleep(300);
+        } catch (InterruptedException e) {}
+      }
+
+      view.getLocationOnScreen(xy);
+      clickOnScreen(xy[0] + view.getWidth() / 2.0f, xy[1] + view.getHeight() / 2.0f);
```

Comment:
Would it be useful to make those two lines a separate function as they're repeated twice and it seems possible to get out of sync if someone changes one without the other.